test/parallel/test-stringbytes-external.js

```diff
@@ -119,3 +122,53 @@ var PRE_3OF4_APEX = Math.ceil((EXTERN_APEX / 4) * 3) - RADIOS;
   assert.equal(a, b);
   assert.equal(b, c);
 })();
+
+// v8 fails silently if string length > v8::String::kMaxLength
+(function() {
+  // v8::String::kMaxLength defined in v8.h
+  var kStringMaxLength = internalBuffer.kStringMaxLength;
+
+  var maxString = new Buffer(kStringMaxLength).toString();
+  assert.equal(maxString.length, kStringMaxLength);
+
+  var maxString = new Buffer(kStringMaxLength).toString('binary');
+  assert.equal(maxString.length, kStringMaxLength);
+
+  var maxString = new Buffer(kStringMaxLength + 1).toString('binary', 1);
+  assert.equal(maxString.length, kStringMaxLength);
+
+  var maxString = new Buffer(kStringMaxLength + 1).toString(
+      'binary', 0, kStringMaxLength);
+  assert.equal(maxString.length, kStringMaxLength);
+
+  var maxString = new Buffer(kStringMaxLength + 2).toString('utf16le');
+  assert.equal(maxString.length, (kStringMaxLength + 2) / 2);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength + 1).toString();
+  }, Error);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength + 1).toString('ascii');
+  }, Error);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength + 1).toString('utf8');
+  }, Error);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength * 2 + 2).toString('utf16le');
+  }, Error);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength + 1).toString('binary');
+  }, Error);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength + 1).toString('base64');
+  }, Error);
+
+  assert.throws(function() {
+    var buf = new Buffer(kStringMaxLength + 1).toString('hex');
+  }, Error);
+})();
```

Comment:
Apropos all the `new Buffer(..).toString(...)` calls, you should probably put a `.fill()` in between to stop valgrind from complaining.  See ac70bc8.